docs: Update Sensor Documentation to Include Tag Retrieval Examples (#28532)
dagster-io/dagster: docs/docs/guides/automate/sensors/index.md

If you want to be able to reset a sensor's cursor, don't set `run_key`s on `RunRequest`s.
:::

> ## Grabbing Tags Inside Sensors

Reviewer: ```suggestion
## Accessing tags from upstream job runs
```

> When working with sensors in Dagster, you might need to access tags from upstream job runs. For example, you can retrieve the partition key from an upstream job's tags. This can be useful for triggering downstream jobs with specific partition keys.

Reviewer: ```suggestion
When working with sensors in Dagster, you might need to access tags from upstream job runs. For example, you may want to retrieve the partition key from an upstream job's tags in order to trigger downstream jobs with specific partition keys:
```

### Example

Here is an example of how to grab the partition key from an upstream job's tags inside an asset sensor:

```python
from dagster import sensor, AssetKey, RunRequest

@sensor(asset_key=AssetKey("my_asset"))
def my_asset_sensor(context, asset_event):
    # Retrieve the partition key from the upstream job's tags
    partition_key = asset_event.dagster_event.logging_tags.get("dagster/partition")
    
    if partition_key:
        # Trigger the downstream job with the partition key
        return RunRequest(
            run_key=partition_key,
            tags={"dagster/partition": partition_key},
        )
    return None

> 

Reviewer: You'll need to add closing ``` here
